build/common.project.props

```diff
@@ -134,6 +134,12 @@
     <VsixPublishDestination>$(ArtifactRoot)$(VsixOutputDirName)\</VsixPublishDestination>
   </PropertyGroup>
 
+  <!-- Write out .XML files for projects that will be packed. -->
+  <PropertyGroup>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <DocumentationFile Condition=" '$(DocumentationFile)' == '' AND '$(GenerateDocumentationFile)' == 'true' AND '$(IsNetCoreProject)' != 'true' ">$(OutputPath)\$(AssemblyName).xml</DocumentationFile>
```

Comment:
I think it's a great thing to enforce XMLDoc for the packages we publish to nuget.org. However,

1. This issue is titled enforce IDE0005, but [the docs say IDE0005 is about removing unnecessary usings](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0005?pivots=lang-csharp-vb), which has nothing to do with XMLDoc, and therefore the title "enforce IDE005 by enabling GenerateDocumentationFile" doesn't make sense
2. I don't see anything in this PR that changes enforcement of IDE005. Our CI builds will continue to pass after this merges if people add new violations in future PRs.
3. To be honest, I'm not a fan of adding NoWarn. I see NoWarn as tech debt, so now that I see the consequence of enabling `GenerateDocumentationFile` to all projects in the solution, I prefer the old implementation where it's only selectively enabled for projects that are packed.